### DISCUSSION TITLE
Monitoring for failed generic app updates

### DISCUSSION THREAD

**Discussion by Shane-MTB** [Romanitho/Winget-AutoUpdate#758] [Q&A]
I'm trying to find a way to monitor for failed app updates with ConnectWise Automate. I was hoping there would be something I could use to trigger a monitor, but I'm struggling to find anything. As far as I can tell the updates.log is the only place anything about the updates is stored. 
I did try modifying the Update-App.ps1 to include Windows Event Log entries, but found that script gets replaced when Winget-AutoUpdate is updated...
Anyone have an idea as to what I could use to trigger a monitor on a failed update?
Thanks!

**Comment by KnifMelti** [comment]
https://github.com/Romanitho/Winget-AutoUpdate?tab=readme-ov-file#custom-scripts-mods-feature-for-apps
Write your own routine in `_WAU-notinstalled.ps1`

**Reply by KnifMelti** [reply]
Under **SYSTEM** context the log files from **WinGet** are written to `C:\Windows\Temp\WinGet\defaultState\`.
The path is only reachable by **SYSTEM**.
When installing an application it creates a log named `WinGet-AppID...`
Parsing them would be a good praxis looking for errors (all MSI installations get an own log file too):

![image](https://github.com/user-attachments/assets/62dc94e3-02d4-4e0e-b591-41b5218d9e68)
